e2etest/bgptests/infra_setup.go

```diff
@@ -147,27 +164,61 @@ func VRFContainersSetup(cs *clientset.Clientset) ([]*frrcontainer.FRR, error) {
 		return nil, errors.Wrapf(err, "failed to create %s: %s", vrfNetwork, out)
 	}
 
+	out, err = executor.Host.Exec(executor.ContainerRuntime, "network", "create", vrfNextHopSettings.multiHopNetwork, "--ipv6",
+		"--driver=bridge", "--subnet=172.32.0.0/16", "--subnet=fc00:f853:ccd:e800::/64")
+	if err != nil && !strings.Contains(out, "already exists") {
+		return nil, errors.Wrapf(err, "failed to create %s: %s", vrfNextHopSettings.multiHopNetwork, out)
+	}
+
 	config := vrfContainersConfig()
 
 	vrfContainers, err := frrcontainer.Create(config)
 	if err != nil {
 		return nil, err
 	}
+
 	err = vrfSetup(cs)
 	if err != nil {
 		return nil, err
 	}
+	err = multiHopSetUp(vrfContainers, vrfNextHopSettings, cs)
+	if err != nil {
+		return nil, err
+	}
+
 	return vrfContainers, nil
 }
 
 // InfraTearDown tears down the containers and the routes needed for bgp testing.
-func InfraTearDown(cs *clientset.Clientset, containers []*frrcontainer.FRR) error {
-	err := frrcontainer.Delete(containers)
+func InfraTearDown(cs *clientset.Clientset) error {
+	return infraTearDown(cs, FRRContainers, defaultNextHopSettings, isVRFContainer)
+}
+
+// InfraTearDown tears down the containers and the routes needed for bgp testing.
+func InfraTearDownVRF(cs *clientset.Clientset) error {
+	return infraTearDown(cs, FRRContainers, vrfNextHopSettings, func(c *frrcontainer.FRR) bool {
+		return !isVRFContainer(c)
+	})
+}
+
+func infraTearDown(cs *clientset.Clientset, containers []*frrcontainer.FRR, nextHop nextHopSettings, filter func(*frrcontainer.FRR) bool) error {
+	multiHopRoutes, err := container.Networks(nextHop.nextHopContainerName)
+	if err != nil {
+		return err
+	}
+
+	filtered := make([]*frrcontainer.FRR, 0)
+	for _, c := range containers {
+		if !filter(c) {
```

Comment:
I think it's more clear doing this the other way (if it matches the filter it is considered)

Reply:
fair enough, but then I'll change the name because filter doesn't sound right.